roles/confluent.kafka_connect/templates/connect-distributed.properties.j2

```diff
@@ -65,113 +51,47 @@ key.converter.schema.registry.url={% for host in groups['schema_registry'] %}{%
 # Connect Producer Configuration
 producer.bootstrap.servers={% for host in groups['kafka_broker'] %}{% if loop.index > 1%},{% endif %}{{ host }}:{{kafka_broker_listeners[kafka_connect_kafka_listener_name]['port']}}{% endfor %}
 
-producer.security.protocol={{kafka_broker_listeners[kafka_connect_kafka_listener_name] | kafka_protocol_defaults(sasl_protocol, ssl_enabled) }}
-{% if kafka_broker_listeners[kafka_connect_kafka_listener_name]['ssl_enabled'] | default(ssl_enabled) | bool %}
-producer.ssl.truststore.location={{kafka_connect_truststore_path}}
-producer.ssl.truststore.password={{kafka_connect_truststore_storepass}}
-{% if kafka_broker_listeners[kafka_connect_kafka_listener_name]['ssl_mutual_auth_enabled'] | default(ssl_mutual_auth_enabled) | bool %}
-producer.ssl.keystore.location={{kafka_connect_keystore_path}}
-producer.ssl.keystore.password={{kafka_connect_keystore_storepass}}
-producer.ssl.key.password={{kafka_connect_keystore_keypass}}
-{% endif %}
-{% endif %}
-{% if kafka_broker_listeners[kafka_connect_kafka_listener_name]['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'PLAIN' %}
-producer.sasl.mechanism=PLAIN
-{% endif %}
-{% if kafka_broker_listeners[kafka_connect_kafka_listener_name]['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'GSSAPI' %}
-producer.sasl.mechanism=GSSAPI
-producer.sasl.kerberos.service.name={{kerberos_kafka_broker_primary}}
-{% endif %}
-{% if kafka_broker_listeners[kafka_connect_kafka_listener_name]['sasl_protocol'] | default(sasl_protocol) | normalize_sasl_protocol == 'SCRAM-SHA-256' %}
-producer.sasl.mechanism=SCRAM-SHA-256
-{% endif %}
+{% set config_prefix = 'producer.' %}
```

Comment:
There's one caveat with `producer.` and `consumer.` property sets - according to the connect team, we should include jaas config for producer and consumer. Currently you import kafka_client.j2 (kudos btw - it is a really cool way to standardize on kafka client properties and avoid copypaste) - can it be made configurable to include sasl by default but allow producing client configs without jaas config?
cc @C0urante - just to check if I'm not missing anything about jaas config

Reply:
@stan-confluent is right; we recommend that each connector be given its own unique principal, so each should include a `sasl.jaas.config` for its Kafka clients in its connector config and there should be no default, global principal in the `connect-distributed.properties` worker config.
I'd err on the side of not including `producer.sasl.jaas.config`, `consumer.sasl.jaas.config`, and `admin.sasl.jaas.config` by default here, but maybe allowing the user to specify that they would like a "global connector principal" configured in their worker file (which would be implemented by adding the aforementioned `*.sasl.jaas.config` properties) if they _really_ want to do things the wrong way.

Reply:
yep... i forgot about this sorry this is the 3rd time @C0urante has reminded me lol im on it

Reply:
ಠ_ಠ

Reply:
so include `sasl.jaas.config`, but do not include `producer.sasl.jaas.config`, `consumer.sasl.jaas.config`, and `admin.sasl.jaas.config` ? @C0urante 

also we do not have admin in this template btw, but i thats a diff issue

Reply:
> so include sasl.jaas.config, but do not include producer.sasl.jaas.config, consumer.sasl.jaas.config, and admin.sasl.jaas.config ?

Yep, exactly

> also we do not have admin in this template btw, but i thats a diff issue

I think this should be fine; we required `admin.*` properties for a few releases but that was due to a bug in the Connect framework (that has since been fixed). If this is targeted strictly for 5.5+, it should be fine as-is without those properties.

Reply:
resolving here: https://github.com/domenicbove/cp-ansible/pull/12